server/utils/hclust.R

```diff
@@ -123,11 +142,35 @@ SortedColumnNames <- colnames(normalized_matrix)[ColumnDend$order]
 #m <- matrix(SortedMatrix,length(SortedRowNames),length(SortedColumnNames))
 #colnames(m) <- SortedColumnNames
 #rownames(m) <- SortedRowNames
-cat("rowindexes",RowDend$order,"\n",sep="\t") # Prints out row indices
-cat("colindexes",ColumnDend$order,"\n",sep="\t") # Prints out column indicies
-cat("rownames",SortedRowNames,"\n",sep="\t") # Prints out row names
-cat("colnames",SortedColumnNames,"\n",sep="\t") # Prints out column names
-cat ("OutputMatrix",normalized_matrix,"\n",sep="\t") # This outputs the 2D array in 1D column-wise. This is later converted to 2D array in nodejs.
+
+output_df <- list()
+output_df$method <- input$cluster_method
+output_df$RowNodeJson <- row_node_json
+output_df$ColNodeJson <- col_node_json
+output_df$RowDendOrder <- {lapply(1:length(RowDend$order), function(y){
+    list(i=RowDend$order[y])
+})}
+output_df$ColumnDendOrder <- {lapply(1:length(ColumnDend$order), function(y){
+    list(i=ColumnDend$order[y])
+})}
+output_df$SortedRowNames <- {lapply(1:length(SortedRowNames), function(y){
+    list(gene=SortedRowNames[y])
+})}
+output_df$SortedColumnNames <- {lapply(1:length(SortedColumnNames), function(y){
+    list(sample=SortedColumnNames[y])
+})}
+output_df$OutputMatrix <- {lapply(1:length(normalized_matrix), function(y){
+    list(elem=normalized_matrix[y])
+})}
+#print ("output_json")
+output_json <- toJSON(output_df)
+print (output_json)
```

Comment:
`toJSON()` will output to standard out. No need for the `print()` statement.

Reply:
I think you need to remove output_json as well when you comment out the print statement. Just "toJSON(output_df)" will work.